examples/defaulter-gen/generators/defaulter.go

```diff
@@ -758,19 +936,31 @@ func (n *callNode) WriteMethod(varName string, depth int, ancestors []*callNode,
 	switch {
 	case n.index:
 		sw.Do("for $.index$ := range $.var$ {\n", vars)
-		if n.elem {
-			sw.Do("$.local$ := $.var$[$.index$]\n", vars)
-		} else {
-			sw.Do("$.local$ := &$.var$[$.index$]\n", vars)
+		if !n.markerOnly {
+			if n.elem {
+				sw.Do("$.local$ := $.var$[$.index$]\n", vars)
+			} else {
+				sw.Do("$.local$ := &$.var$[$.index$]\n", vars)
+			}
 		}
 
+		n.writeDefaulter(varName, index, isPointer, sw)
 		n.writeCalls(local, true, sw)
 		for i := range n.children {
 			n.children[i].WriteMethod(local, depth+1, append(ancestors, n), sw)
 		}
 		sw.Do("}\n", nil)
 	case n.key:
+		if n.defaultValue != "" {
+			// Map keys are typed and cannot share the same index variable as arrays and other maps
+			index = index + "_" + ancestors[len(ancestors)-1].field
```

Comment:
I think this is the best way to guarantee index variables are unique for map iterations. It's impossible for two variables under the same struct to share the same name